e2e/README.md

```diff
@@ -1,101 +1,68 @@
-`go mod tidy`:
+We will be using a kind cluster and install CAPL plus various other providers.
 
-```
-cd ~/go/src/github.com/linode/linode-blockstorage-csi-driver
-go mod tidy
-```
-
-Then, use the makefile in the directory above this directory to build the CSI
-(this is to download goimports)
+Run the following command to create a kind mgmt cluster:
 
+```sh
+devbox run local-deploy
 ```
-cd $GOPATH/src/github.com/linode/linode-blockstorage-csi-driver
-make build
-```
-
-### Running
-
-Build a docker image passing the `IMAGE_TAG` argument to the make target
-so a custom tag is applied. Then push the image to a public repository.
+This will download all the necessary binaries to local bin and create a local mgmt cluster.
 
-> You can use any public repository that you have access to. The tags used below are just examples
+### 2. Build and Push Test Image
 
-```
-make docker-build IMAGE_TAG=ghcr.io/avestuk/linode-blockstorage-csi-driver:test-e2e
-make docker-push IMAGE_TAG=ghcr.io/avestuk/linode-blockstorage-csi-driver:test-e2e
-```
+If you have a PR open, GHA will build & push to docker hub and tag it with the current branch name.
 
-Finally run the tests passing the name of the image and the tag so that your custom image is used.
-```
-make test IMAGE_NAME=ghcr.io/avestuk/linode-blockstorage-csi-driver IMAGE_TAG=test-e2e
-```
+If you do not have PR open, follow the steps below:
+- Build a docker image passing the `IMAGE_TAG` argument to the make target
+  so a custom tag is applied. Then push the image to a public repository.
 
-When running the e2e tests, a couple of options can be passed to the test
-through the `$SUITE_ARGS` environment variable to modify its behavior: 
+  > You can use any public repository that you have access to. The tags used below are just examples
 
- - If the Linode API extra debugging logs are desired, simply use the
-   `--linode-debug` in the list of suite arguments. NOTE: This will also
-   print out the Linode API token in the logs, since they will be part of the
-   requests being logged.
+  ```
+  make docker-build IMAGE_TAG=ghcr.io/avestuk/linode-blockstorage-csi-driver:test-e2e
+  make docker-push IMAGE_TAG=ghcr.io/avestuk/linode-blockstorage-csi-driver:test-e2e
+  ```
 
- - Similarly, the Linode API base URL can be changed from
-   `https://api.linode.com` with the `--linode-url` flag.
+### 2. Setup a CAPL Child Test Cluster
 
-#### Creating a new cluster
+In order create a test cluster, setup the env vars and run the following command:
 
-By default the tests use $HOME/.ssh/id\_rsa.pub as the public key used to
-provision the cluster, so it needs to be added to your agent.
+```sh
+# Make sure to set the following env vars
+export LINODE_TOKEN="your linode api token"
+export REGION="your preferred region"
+export KUBERNETES_VERSION=v1.29.1
+export LINODE_CONTROL_PLANE_MACHINE_TYPE=g6-standard-2
+export LINODE_MACHINE_TYPE=g6-standard-2
```

Comment:
I think we also need WORKER_MACHINE_COUNT. Or do we want to just have control plane node and no worker nodes?

Reply:
As of now, I have it configured to just run on the control plane node. I tweaked the tolerations so statefulsets and pods would be scheduled on the control plane node for testing. Do you think we should add worker node for testing?

Reply:
I think thats fine. Adding a worker node is going to slow down tests by another 3-4 mins, so we might be good if we just have one control plane machine to test the csi stuff. Looks like we are not doing tests which attach/detach/reattach volumes to different nodes, so might be good here.